### ISSUE TITLE
Emit close never fired but actually finished unzipping

### ISSUE THREAD

**Issue by toshim45** [ZJONSSON/node-unzipper#132]
From your readme:
> Extract emits the 'close' event once the zip's contents have been fully extracted to disk. Extract uses fstream.Writer and therefore needs need an absolute path to the destination directory. This directory will be automatically created if it doesn't already exits.

My snippet:
```
console.log('parser-debug: unzip', id, fileName)

let unzipExtractor = unzipper.Extract({path: OfflineSync.ParserPath,verbose: false})
	
unzipExtractor.on('error', (error)=>{
	console.log('parser-error: unzip', id, fileName, error)
})

unzipExtractor.on('close', ()=>{
	console.log('parser-debug: zip-close-parse', id, fileName)
})

fs.createReadStream(OfflineSync.UploadPath + fileName).pipe(unzipExtractor)
```

result:

```
macko-ss:parserv2 artikow$ node parser.js 
parser-debug: unzip 5717 Sarang_20190527132211.zip
macko-ss:parserv2 artikow$ 
```

it seems close event never fired, but unzip process is actually finished, file extracted.

I use "unzipper": "^0.9.15" and node v8.9.4

Thanks

**Comment by toshim45** [comment]
hmm ok I think you're still working on something on the extract.js 5e26471cc7f70e520e168b2a781f447459558d7f
this version 0.9.15 i think is changeset be0f2a412819bd82207e1af3e94776d16ca8d9b3

**Comment by ZJONSSON** [comment]
Can you try with master? (which currently is unpublished but slated for a minor bump)
`npm install zjonsson/unzipper#master`

**Comment by toshim45** [comment]
> npm install zjonsson/unzipper#master

got permission error

```
npm ERR! Error while executing:
npm ERR! /usr/bin/git ls-remote -h -t ssh://git@github.com/zjonsson/unzipper.git
npm ERR! 
npm ERR! Warning: Permanently added 'github.com,13.229.188.59' (RSA) to the list of known hosts.
npm ERR! Permission denied (publickey).
npm ERR! fatal: Could not read from remote repository.
```

**Comment by ZJONSSON** [comment]
oops sorry I got the name wrong, try `npm install zjonsson/node-unzipper#master`

**Comment by toshim45** [comment]
its working 👍 thanks, will wait for next publish

**Comment by ZJONSSON** [comment]
published as `+ unzipper@0.10.0`.  Please reopen if issue is not resolved
